src/nemo-devicelock/host/mcedevicelock.cpp

```diff
@@ -379,4 +379,41 @@ void MceDeviceLockAdaptor::setState(int state)
     }
 }
 
+void MceDeviceLockAdaptor::activateTemporaryLockout()
+{
+    if (m_deviceLock->automaticLocking() == -1) {
+        m_deviceLock->sendErrorReply(QDBusError::AccessDenied, QStringLiteral("Device lock not in use"));
+        return;
+    }
+
+    qint64 temporaryLockTimeout = m_deviceLock->temporaryLockTimeout();
+    if (temporaryLockTimeout <= 0 || temporaryLockTimeout > 24 * 60 * 60) {
```

Comment:
We could have `TEMPORARY_LOCK_MAX_TIMEOUT` ?

Reply:
Yeah. The same with attempt counters. Though I'm struggling to find a place where to put such constants. These values are sort owned by backend, so that would seem logical. But that is by design not visible to common logic in here. Using dynamic limits = new method might work, but that would be an avalanche of interface changes. And Settings UI is using its set of hard coded limits anyway...

But even defining the same / similar constant in several places would provide logical context or sorts and thus would be better than raw numbers.

Reply:
After a cup of coffee. The worry / paranoia is that there is a bit of history of problems due to having storage(s) / backends / common parts using mixed widths and differing signedness types... In this case storage cutting off high bits is not a problem. So it leaves possible int vs uint issues. Which would be uint processing misinterpreting INT_MAX to UINT_MAX range. Here we are using signed values and overflows elsewhere would show up as negatives -> high bound check is not actually needed, or?

Reply:
Removed upper bound checks